### ISSUE TITLE
Regular debug build broken

### ISSUE THREAD

**Issue by jjbandit** [oilshell/oil#1191]
Looks like `ninja _bin/cxx-dbg/osh-eval` is complaining about multiple output paths for some reason?  I took a look at the ninja file but figured you'd probably be more effective at figuring this one out @andychu .

If I try to build with `build/dev.sh oil-cpp` (when forcing the non-clang build) I get the following output:

```
  ── ! 1 job oil build/dev.sh oil-cpp                                                                                                    ±[A3**][master]
asdl/hnode.asdl -> _build/cpp/hnode_asdl* and _devbuild/gen/hnode_asdl_debug.py
frontend/types.asdl -> _build/cpp/types_asdl* and _devbuild/gen/types_asdl_debug.py
-rw-rw-r-- 1 scallywag scallywag 22811 Jul  2 06:35 _build/cpp/id_kind_asdl.cc
-rw-rw-r-- 1 scallywag scallywag 12859 Jul  2 06:35 _build/cpp/id_kind_asdl.h
-rw-rw-r-- 1 scallywag scallywag 46805 Jul  2 06:35 _build/cpp/consts.cc
-rw-rw-r-- 1 scallywag scallywag  1318 Jul  2 06:35 _build/cpp/consts.h
  (core/optview_gen) -> _build/cpp/core_optview.h
core/runtime.asdl -> _build/cpp/runtime_asdl* and _devbuild/gen/runtime_asdl_debug.py
frontend/syntax.asdl -> _build/cpp/syntax_asdl* and _devbuild/gen/syntax_asdl_debug.py


arith_parse_gen.py: precedence table has 47 entries
-rw-rw-r-- 1 scallywag scallywag 13739 Jul  2 06:35 _build/cpp/arg_types.cc
-rw-rw-r-- 1 scallywag scallywag 11977 Jul  2 06:35 _build/cpp/arg_types.h
./NINJA_config.py: Wrote _build/oil-native.sh
-rwxrwxr-x 1 scallywag scallywag 6702 Jul  2 06:35 _build/oil-native.sh
	 Before importing: 44 modules
	 Importing 'bin.osh_eval'
	 After importing: 255 modules
  93 _build/app-deps/osh_eval/all.txt
  52 _build/app-deps/osh_eval/translate.txt
  77 _build/app-deps/osh_eval/typecheck.txt
 222 total
	mycpp: LOADING asdl/format.py asdl/runtime.py bin/osh_eval.py core/alloc.py core/ansi.py core/dev.py core/error.py core/executor.py core/main_loop.py core/process.py core/shell_native.py core/state.py core/ui.py core/util.py core/vm.py frontend/args.py frontend/lexer.py frontend/location.py frontend/parse_lib.py frontend/reader.py frontend/typed_args.py oil_lang/expr_parse.py oil_lang/expr_to_ast.py oil_lang/funcs.py oil_lang/regex_translate.py osh/arith_parse.py osh/bool_parse.py osh/braces.py osh/builtin_assign.py osh/builtin_bracket.py osh/builtin_meta.py osh/builtin_misc.py osh/builtin_printf.py osh/builtin_pure.py osh/cmd_eval.py osh/cmd_parse.py osh/glob_.py osh/prompt.py osh/sh_expr_eval.py osh/split.py osh/string_ops.py osh/tdop.py osh/word_.py osh/word_compile.py osh/word_eval.py osh/word_parse.py pgen2/grammar.py pgen2/pnode.py pylib/cgi.py pylib/os_path.py qsn_/qsn.py qsn_/qsn_native.py
	mycpp pass: CONST
	mycpp pass: FORWARD DECL
	mycpp pass: PROTOTYPES
	mycpp pass: IMPL

real	0m14.659s
user	0m14.537s
sys	0m0.116s
./NINJA_config.py: Wrote build.ninja
ninja: error: _build/obj/cxx-dbg/core_pyutil.o.d: depfile has multiple output paths
```

**Comment by andychu** [comment]
Oh sorry, I think that was broken at one point yesterday (maybe on `soil-staging`), but not on `master` right now?

I just did 

```
git checkout master
git pull
build/native.sh clean
build/dev.sh oil-cpp
ninja _bin/cxx-db/osh_eval
```

and it worked.

(BTW I think we should get rid of the `compile-quickly` thing in `build/dev.sh oil-cpp`; it's more trouble that it's worth now.  It can still be an option)

**Comment by andychu** [comment]
Specifically I think it was broken before this commit: https://github.com/oilshell/oil/commit/93dc736d1b11e7b0138a8d41a98ed55e7e3f6682

If you ran a build there, it got the arguments mixed up, and could have written a corrupted `.d`.    Sorry about that

**Comment by jjbandit** [comment]
Ah yeah, I had to run `build/native.sh clean` for some reason.  Guess there must have been a corrupt build file somewhere ..?

**Comment by jjbandit** [comment]
Thanks :D

**Comment by andychu** [comment]
Also I was trying to figure out if there was a race condition in the merge bot that let this commit through ... If it fails any tests (and this did), it should never appear on `master`.

But I thought there might be a race, and then I looked at the `soil/maybe-merge.sh` code and it seems correct.  (There's no way to tell now since I had to destroy and recreate the `soil-staging` branch ...)

But be on the lookout for anything broken on `master`.

Today I'm going to unify the ninja build a bit more and move stuff around, but it should stabilize soon and https://github.com/oilshell/oil/wiki/Oil-Dev-Cheat-Sheet will be kept up to date
